src/diagonal.works/b6/api/functions/change.go

```diff
@@ -107,6 +108,65 @@ func addRelation(c *api.Context, id b6.RelationID, tags api.TagCollection, membe
 	}, nil
 }
 
+func addCollection(c *api.Context, id b6.CollectionID, tags api.TagCollection, collection api.Collection) (ingest.Change, error) {
+	feature := &ingest.CollectionFeature{
+		CollectionID: id,
+	}
+
+	i := tags.Begin()
+	for {
+		ok, err := i.Next()
+		if err != nil {
+			return nil, err
+		} else if !ok {
+			break
+		}
+		tag, ok := i.Value().(b6.Tag)
+		if !ok {
+			return nil, fmt.Errorf("Expected b6.Tag, found %T", i.Value())
+		}
+		feature.Tags = append(feature.Tags, tag)
+	}
+
+	i = collection.Begin()
+	for {
+		ok, err := i.Next()
+		if err != nil {
+			return nil, err
+		} else if !ok {
+			break
+		}
+
+		feature.CollectionProto.Keys = append(feature.CollectionProto.Keys, toLiteralNodeProto(i.Key()))
+		feature.CollectionProto.Values = append(feature.CollectionProto.Values, toLiteralNodeProto(i.Value()))
+	}
+
+	return &ingest.AddFeatures{
+		Collections: []*ingest.CollectionFeature{feature},
+	}, nil
+}
+
+func toLiteralNodeProto(v interface{}) *pb.LiteralNodeProto {
+	switch x := v.(type) {
+	// TODO: cover all types
+	case b6.Identifiable:
+		return &pb.LiteralNodeProto{Value: &pb.LiteralNodeProto_FeatureIDValue{
+			FeatureIDValue: &pb.FeatureIDProto{
+				Type:      pb.FeatureType_FeatureTypeCollection,
```

Comment:
not sure it makes sense to have feature type collection here ? b/c like is that for when we have collections of collections ?

Reply:
I think these are fine for the moment, and we can expand later as needed, but see comments below on breaking the dependency on protocol buffers, at which point, I think the `collectionFeature` struct can just include `keys interface{}` and `values interface{}` where those are slices.